src/routes/wohnen/containers/WohnenExtra.js

```diff
@@ -10,38 +10,70 @@ import WohnenOfferModel from '../../../modules/endpoint/models/WohnenOfferModel'
 import CityModel from '../../../modules/endpoint/models/CityModel'
 import ExtraModel from '../../../modules/endpoint/models/ExtraModel'
 import OfferList from '../components/OfferList'
+import Offer from '../components/Offer'
+import Hashids from 'hashids'
 
 type PropsType = {
-  offers: Array<WohnenOfferModel>,
+  offers: Array<WohnenOfferModel<*>>,
   city: string,
   language: string,
+  offerHash: string,
   extras: Array<ExtraModel>,
   cities: Array<CityModel>
 }
 
 class WohnenExtra extends React.Component<PropsType> {
-  render () {
-    const LoadingSpinner = () => <Spinner name='line-scale-party' />
+  hashids = new Hashids()
+  hash = (offer: WohnenOfferModel<*>) => this.hashids.encode(offer.email.length, offer.createdDate.seconds())
+
+  findOfferByHash (hash: string): WohnenOfferModel<*> | void {
+    return this.props.offers.find(offer => this.hash(offer) === hash)
+  }
 
-    const {offers, extras, cities, city} = this.props
+  render () {
+    const {offers, extras, cities, city, language, offerHash} = this.props
     const cityName = CityModel.findCityName(cities, city)
     const extra: ExtraModel | void = extras.find(extra => extra.alias === 'wohnen')
 
     if (!extra) {
       return null
     }
 
+    if (!offers) {
+      return <Spinner name='line-scale-party' />
+    }
+
+    if (offerHash) {
+      const offer = this.findOfferByHash(offerHash)
+
+      if (!offer) {
+        return <div>Offer not found!</div>
+      }
+
+      return (
+        <React.Fragment>
```

Comment:
We should display a `<Caption />` like in the Sprungbrett-Extra.
I'd suggest `<Caption title='Raumfrei' />` :D

Reply:
Already there, but refactored to be more clear :)